src/mongo/db/catalog/index_catalog_impl.cpp

```diff
@@ -813,8 +814,8 @@ Status IndexCatalogImpl::_doesSpecConflictWithExisting(OperationContext* opCtx,
                                             << " already exists with different options: "
                                             << desc->infoObj());
 
-            return Status(ErrorCodes::IndexAlreadyExists,
-                          str::stream() << "index already exists with different name: " << name);
+            return Status(ErrorCodes::CannotCreateIndex,
+                          str::stream() << "Index with the same keys but a diferent name already exists.");
```

Comment:
Please use 'options' instead of 'keys' here. To me, 'keys' implies the object had the same field names, but not necessarily the same values for those field names.